src/rgw/rgw_putobj_processor.cc

```diff
@@ -145,14 +252,14 @@ int AtomicObjectProcessor::prepare(optional_yield y)
 
   rgw_raw_obj stripe_obj = manifest_gen.get_cur_obj(store);
 
-  r = writer->set_stripe_obj(stripe_obj);
+  r = writer.set_stripe_obj(stripe_obj);
   if (r < 0) {
     return r;
   }
 
   set_head_chunk_size(head_max_size);
   // initialize the processors
-  chunk = ChunkProcessor(writer.get(), chunk_size);
+  chunk = ChunkProcessor(&writer, chunk_size);
```

Comment:
Does this and related code imply that non-Rados stores have to care about (Rados-based) chunks/stripes?

Reply:
(Because I think that should be the case)

Reply:
No.  putobj_processor is now officially part of RadosStore only, and can be safely ignored by other stores.  That's the point of moving DataProcessor into rgw::sal, it becomes the new API.